app/src/ui/changes/commit-message.tsx

```diff
@@ -628,17 +673,26 @@ export class CommitMessage extends React.Component<
     const { isCommitting, branch: branchName, commitButtonText } = this.props
 
     const loading = isCommitting === true ? <Loading /> : undefined
-    const commitVerb = loading ? 'Committing' : 'Commit'
+
+    const isAmending = this.props.commitToAmend !== null
+    const commitVerb = isAmending
+      ? loading
+        ? 'Amending'
+        : 'Amend'
+      : loading
+      ? 'Committing'
+      : 'Commit'
     const commitTitle =
       branchName !== null ? `${commitVerb} to ${branchName}` : commitVerb
-    const defaultContents =
-      branchName !== null ? (
-        <>
-          {commitVerb} to <strong>{branchName}</strong>
-        </>
-      ) : (
-        commitVerb
-      )
+    const defaultContents = isAmending ? (
+      <>{commitVerb} last commit</>
+    ) : branchName !== null ? (
+      <>
+        {commitVerb} to <strong>{branchName}</strong>
+      </>
+    ) : (
+      commitVerb
+    )
```

Comment:
Suggestion. I found this hard to read.. maybe worth separating into two variables. commitDefaultContents and amendDefaultContents and then defaultContents = isAmending ? amendDefaultContents : commitDefaultContents